src/tests/EShopworld.WorkerProcess.UnitTests/Infrastructure/SystemTimerTests.cs

```diff
@@ -10,43 +10,57 @@ namespace EShopworld.WorkerProcess.UnitTests.Infrastructure
 {
     public class SystemTimerTests
     {
+        
         [Fact, IsUnit]
-        public async Task SystemTimer_WhenLoopStarts_ExecutorShouldRun()
+        public void SystemTimer_WhenLoopStarts_ExecutorShouldRun()
```

Comment:
I don't think this test will work? The token is canceled when you execute your timer so it will not call the lambda?

Reply:
Yes, the idea behind this test is that if we stop the timer no lambda execution should be performed. Can you please clarify why do you think it wouldn't work?

Reply:
Because you have `isCalled.Should().BeTrue();` which will fail in my view.
It will fail because your lambda will never get executed.
It will not be executed because you line 25 in SystemTimer.cs

Reply:
Did you run it locally?

Reply:
sorry you are actually referirng to the test that **should** invoke the lambda. The assertion is correct as I'm expect in this case the lambda to be called. will try to describe what happens
 var task=timer.ExecutePeriodicallyIn(TimeSpan.FromMilliseconds(1), async (token)=>
            {
                isCalled = true;
                manualReset.Set();
                return await Task.FromResult(TimeSpan.Zero);
            });
//Here the timer starts a new task for an endless loop to the lambda periodically
  manualReset.WaitOne(TimeSpan.FromMilliseconds(200));
   timer.Stop();
//Here i wait until the manualreset is set which happens after isCalled is set to true then i stop the timer. Only at that point i can await the task with this  call Func<Task> act = async () => await task; to make sure that at this point the endless loop has being stopped. Hope it clarifies now

Reply:
I stand corrected - seems the lack of await is not getting simply a Task returned, instead it kicks it off, so the test should work in principle.